---
title: "The role of DNA methylated genes in bladder cancer tumor recurrence"
author: "Jess Byun, Michael Jetsupphasuk, JP Flores, Eric Brown"
date: "`r format(Sys.time(), '%m/%d/%Y')`"
output:
  html_document: default
  pdf_document: default
editor_options:
  markdown:
    wrap: 72
---

1.  **Abstract**

2.  **Introduction**

3.  **Materials and Methods**

I.  Research Participants

II. DNA methylation data pre-processing

```{r message=FALSE}
# We download data from cbiobank data_mrna_seq_v2gz, data_clinical_patient.txt.gz and methylation data from GDC portal.
library(tidyr)
library(dplyr)
library(data.table)
library(limma)
library(bumphunter)
library(biomaRt)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(GenomicFeatures)
library(tidyverse)

#Loading all datas
dir <- c("/pine/scr/s/e/seyoun/project_784/Comp_Bio_784_FinalProject/01.data")
rsem <- fread(paste0(dir,"/data_mrna_seq_v2_rsem.txt.gz")) |> as.data.frame()
patient <- fread(paste0(dir,"/data_clinical_patient.txt.gz")) |> as.data.frame()
methyl <- fread(paste0(dir,"/TCGA-BLCA_DNAmethylation450k_Methyl.txt_v2.gz")) |> as.data.frame()
```

```{r}
# This steps were only selecting the samples in "Tumor-free and "Tumor"  
methly_beta <- methyl[,-c(1:4)]
methyl_nodup <- as.data.frame(sapply(unique(names(methly_beta)),function(col) rowMeans(methly_beta[names(methly_beta) == col])))

colnames(rsem) <- c(colnames(rsem)[1:2],gsub('.{3}$',"",colnames(rsem)[-c(1:2)]))
tumor_status <- c("TUMOR FREE","WITH TUMOR")
patient_subset <- patient[,c(2,21)][is.element(patient[,21],tumor_status),]

sub_methyl <- cbind(methyl[,1:4],methyl_nodup[,(is.element(colnames(methyl_nodup), patient_subset[,1]))])
sub_rsem <- cbind(rsem[,1:2],rsem[,(is.element(colnames(rsem), patient_subset[,1]))])

subset_nm <- colnames(sub_methyl[,!is.element(colnames(sub_methyl),colnames(sub_rsem))])[-c(1:4)]
#three samples are not overlapping with methylation and gene expression. Therefore, to match up, we subtract "TCGA-XF-A9SG" "TCGA-ZF-A9RG" "TCGA-XF-AAMF".

sub_methyl_2 <-sub_methyl[,!is.element(colnames(sub_methyl),subset_nm)]

# This step is removing probes with row containing missing values: 102,702 probes are
sub_methyl_3 <- sub_methyl_2 %>% drop_na()
```

```{r message=FALSE}
#Finding the promoter region -1500, +500 from TSS
library(biomaRt)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(GenomicFeatures)
library(tidyverse)
library(ChAMP)

#ensembl=useMart("ensembl")
#ensembl = useDataset("hsapiens_gene_ensembl",mart=ensembl)
#save(ensembl,file="ensembl"

#If you havent make the ensemble, please run useMART and create ensemble
load("ensembl")
sub_methyl_3[,"Chromosome"] <- sub("^","chr",sub_methyl_3[,"Chromosome"])
methyl.ran <- GRanges(Rle(sub_methyl_3[,"Chromosome"]),IRanges(sub_methyl_3[,"Genomic_Coordinate"],sub_methyl_3[,"Genomic_Coordinate"]))

#We are using hg19
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene

#promoter region set up to 1500, 500
promoter.re <- promoters(txdb,upstream=1500,downstream=500)


fi.cns <- c("TXID","TXCHROM","TXNAME","GENEID","TXSTART","TXEND","TXSTRAND")
txTable <- gsub(" ","",as.matrix(biomaRt::select(txdb,keys=names(promoter.re),columns=fi.cns,keytype="TXNAME")))
txTable <- txTable[!is.na(txTable[,2]),]

map.id <- getBM(attributes=c("hgnc_symbol","entrezgene_id"),filters="entrezgene_id", values=txTable[,"GENEID"],mart=ensembl)
mer.txTable <- unique(as.matrix(merge(txTable,map.id,by.x="GENEID",by.y="entrezgene_id")))
rownames(mer.txTable) <- mer.txTable[,"TXNAME"]


inter.pro.methyl <- as.matrix(findOverlaps(methyl.ran,promoter.re))
inter.pro.methyl <- inter.pro.methyl[is.element(names(promoter.re)[inter.pro.methyl[,2]],rownames(mer.txTable)),]
pro.met.re <- cbind(mer.txTable[names(promoter.re)[inter.pro.methyl[,2]],"hgnc_symbol"],sub_methyl_3[inter.pro.methyl[,1],c(1,3,4)])
colnames(pro.met.re) <- c("gene","methyl","chr","pos")
pro.met.re <- unique(pro.met.re[which(pro.met.re[,"gene"] != ""),])

#Final pro.met.re contains gene name, methylation id, chromosome, and position. 

```

III. Differential methylation analysis


```{r message=FALSE}
#Bumphunter()
library(minfi)
library(MultiDataSet)

promoter_met_mat <- sub_methyl_3[(is.element(sub_methyl_3[,"Hybridization-REF"],pro.met.re[,"methyl"])),]
rownames(promoter_met_mat) <- promoter_met_mat[,1]
mat <- promoter_met_mat[,-c(1:4)] %>% as.matrix()
grset <- makeGenomicRatioSetFromMatrix(mat)
patient_coldata <- patient[is.element(patient[,21],tumor_status),]
patient_coldata <- patient_coldata[(!is.element(patient_coldata[,2],subset_nm)),]
colnames(patient_coldata) <- gsub("\\s", "", colnames(patient_coldata))
patient_coldata[,21] <- gsub("\\s", "", patient_coldata[,21])
patient_coldata$NoninvasiveBladderCancerTreatmentType %<>% gsub("\\(|)", "", .) %>% gsub("\\[|]","",.) %>% gsub("\\s", "",.)
colData(grset) <- cbind(colData(grset),patient_coldata[ order(match(patient_coldata[,2],rownames(colData(grset)))), ])

status <- pData(grset)$PersonNeoplasmStatus
treatment <- pData(grset)$NoninvasiveBladderCancerTreatmentType
sex <-  pData(grset)$Sex
age <- pData(grset)$DiagnosisAge

deseignmatrix <- model.matrix(~status+sex+age)
#we use sex and age for covariates and 10% difference between the two groups

dmrs <- bumphunter(grset, design =deseignmatrix, type="Beta",cutoff=0.1)
head(dmrs$table)
```


```{r}

```

IV. Functional enrichment and network analysis

```{r}

```


4.  **Results**

    I.  **Overview of the study cohort**

    II. **Differential DNA methylation analysis in the two subgroups**

        i.  **Linear**

        ii. **Bumphunter**

            a.  **Minfi**

            b.  **Champ**

5.  **Discussion**

    I.  **Recap**

    II. **Observation**

    III. **Limitation**

    IV. **Future work**

6.  **Conclusion**

7.  **References**

### 
