---
title: "The role of DNA methylated genes in bladder cancer tumor recurrence"
author: "Jess Byun, Michael Jetsupphasuk, JP Flores, Eric Brown"
date: "`r format(Sys.time(), '%m/%d/%Y')`"
output:
  html_document: default
  pdf_document: default
editor_options:
  markdown:
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```

```{r}
# load packages
library(tidyr)
library(data.table)
library(limma)
library(bumphunter)
library(biomaRt)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(GenomicFeatures)
library(tidyverse)
library(ChAMP)
library(readr)
library(minfi)
library(MultiDataSet)
library(ggpubr)

# set ggplot theme
theme_set(theme_bw())
```

# Abstract

Despite advances in bladder cancer treatment, cancer mortality rates have not changed significantly over decades. One of the critical clinical signs of mortality and recurrence is muscle invasiveness. Eighty percent of bladder cancer is due to nonmuscle invasive bladder cancer (NMBIC), and this subgroup has a high recurrence rate, but muscle-invasive bladder cancer (MIBC) has been studied less. Therefore, this study aimed to examine bladder cancer tumor recurrence in a broader spectrum to identify methylation changes. Using two different linear model algorithms (i.e., limma and bumphunter), we identified 128 and 15 differentially methylated genes, respectively. We found that these combined gene set significantly affected the VEGF signaling pathway. These results suggest opportunities for better treatment strategies for bladder cancer recurrences on molecular features. 

# Introduction

Bladder cancer is a complex phenotype dependent on a variety of factors, including genetics [1]–[3], exposure to environmental chemicals [4], [5], and psychosocial stressors. Despite advances in bladder cancer treatment over the past two decades, bladder cancer mortality has not decreased at comparable rates relative to other cancers, such as prostate, breast, lung, and colon cancer [6]. Disparities are stark in bladder cancer incidence rates, as men are 4 times more at risk of being affected by bladder cancer than women [7]. Bladder cancer is classified into two big categories based on tumor invasion. One is the most common form, nonmuscle invasive bladder cancer (NMIBC), which occurs in about 70-80% of bladder cancer cases, and muscle-invasive bladder cancer (MIBC), which occurs in about 20-30% [8]. Due to a large portion of the incidence being due to NMIBC, there is increasing evidence supporting recurrence changes due to chromosome changes and gene mutation [9]. Although MIBC still has shown the incidence of one-third of bladder cancer, molecular profiling has been less studied. Therefore, it is important to understand the broad spectrum of bladder cancer recurrence in genetic components. 

Epigenetic alterations have been suggested to be critical to the underlying etiology of bladder cancer and bladder cancer subtypes [10], [11]. Epigenetic alterations, specifically DNA methylation, may conspire with genetic alterations to induce the bladder cancer phenotype and its subsequent recurrence [12]. Previous research has not identified a bonafide DNA methylation biomarker for bladder cancer, specifically for tumor recurrence. The objective of this study was to identify methylation differences in bladder cancer recurrence based on genomic alteration (tumor status) and to consider how the differentially methylated genes may change the gene expression. 

# Materials and Methods

## Research Participants

The methylation array pre-processed beta value was available via the GDC data portal (Data release 35.0) [13], and RSEM gene expression and clinical patient data were available from cBioPortal [14].

We reviewed the clinical and genetic tumor data of 412 patients with bladder urothelial cancer obtained from The Cancer Genome Atlas (TCGA). We excluded patients with missing tumor status and selected only patients where RSEM gene expression was available. Of the 412 patients, 368 patients were considered in this analysis.

The below table summarizes some summary statistics of the study sample. Patients in the sample tended to be older, male, and tumor free. 

<center>

| Variables  | Count (n=368) |
|------------|:-------------:|
| Age        |               |
| \< 65      | 136 (36.96%)  |
| \>= 65      | 232 (63.04%)  |
| Sex        |               |
| Female     |  94 (25.54%)  |
| Male       | 274 (74.46%)  |
| Status     |               |
| Tumor Free | 234 (63.59%)  |
| With Tumor | 134 (36.41%)  |

**Table 1**: Clinical features of patients with bladder urothelial
carcinoma in TCGA

</center>


## DNA methylation data pre-processing

Bladder cancer recurrence was defined based on the tumor status
previously described by Shen et al. [15] In this study, we classified the
recurrence status based on the neoplasm tumor status column in the
patient data. The term "With Tumor" represents the persistence of the
tumor, but not a new tumor event, and is classified as disease
recurrence. The "Tumor Free" status was classified as having no disease
recurrence. It is important to note that "tumor-free" does not mean
"normal" because the "normal" case implies that the tumor never occurred
in the past and/or present. Furthermore, "tumor-free" specifies that the
tumor cannot be measured and that there is still a chance of recurrence.

The given methylation level was quantified and calculated as $\beta=\frac{\mu}{(M+U+\alpha)}$ where M and U denote the methylated and unmethylated signals, respectively, and $\alpha$ represents an offset value. The given gene expression level was quantified by TPM and normalized using the `RSEM` R package, such that $TPM_{i}=\frac{q_{i}/l_{i}}{\sum_{j}(q_{i}/l_{i})} * 10^{6}$ where $q_{i}$ denotes reads mapped to transcripts and $l_{i}$ is the transcript length. The gene expression level was mapped to the hg19 genome and generated RSEM gene expression data. The promoter region was defined as 1500bp upstream and 500bp downstream from the transcription start site (TSS). A total of 21,010 genes (164,113 methylation probes) were tested. The below figure shows how the promoter regions were defined.

<center>

![](image/promoter_region.png)

**Figure 1**: A schema methylation proves within the promoter region.
Methylation probes between -1500 and +500bp from the transcription start
site are considered.

</center>

```{r}
# We download data from cbiobank data_mrna_seq_v2gz, data_clinical_patient.txt.gz and methylation data from GDC portal.
# loading data
rsem <- fread("01.data/data_mrna_seq_v2_rsem.txt.gz") |> as.data.frame()
patient <- fread("01.data/data_clinical_patient.txt.gz") |> as.data.frame()
methyl <- fread("01.data/TCGA-BLCA_DNAmethylation450k_Methyl.txt_v2.gz") |> as.data.frame()
```

```{r}
# This steps were only selecting the samples in "Tumor-free and "Tumor"  
methly_beta <- methyl[,-c(1:4)]
methyl_nodup <- as.data.frame(sapply(unique(names(methly_beta)),function(col) rowMeans(methly_beta[names(methly_beta) == col])))

colnames(rsem) <- c(colnames(rsem)[1:2],gsub('.{3}$',"",colnames(rsem)[-c(1:2)]))
tumor_status <- c("TUMOR FREE","WITH TUMOR")
patient_subset <- patient[,c(2,21)][is.element(patient[,21],tumor_status),]

sub_methyl <- cbind(methyl[,1:4],methyl_nodup[,(is.element(colnames(methyl_nodup), patient_subset[,1]))])
sub_rsem <- cbind(rsem[,1:2],rsem[,(is.element(colnames(rsem), patient_subset[,1]))])

subset_nm <- colnames(sub_methyl[,!is.element(colnames(sub_methyl),colnames(sub_rsem))])[-c(1:4)]
#three samples are not overlapping with methylation and gene expression. Therefore, to match up, we subtract "TCGA-XF-A9SG" "TCGA-ZF-A9RG" "TCGA-XF-AAMF".

sub_methyl_2 <- sub_methyl[,!is.element(colnames(sub_methyl),subset_nm)]

# This step is removing probes with row containing missing values: 102,702 probes are
sub_methyl_3 <- sub_methyl_2 %>% drop_na()
```

```{r message=FALSE}
#Finding the promoter region -1500, +500 from TSS

ensembl=useMart("ensembl")
ensembl = useDataset("hsapiens_gene_ensembl",mart=ensembl)

sub_methyl_3[,"Chromosome"] <- sub("^","chr",sub_methyl_3[,"Chromosome"])
methyl.ran <- GRanges(Rle(sub_methyl_3[,"Chromosome"]),IRanges(sub_methyl_3[,"Genomic_Coordinate"],sub_methyl_3[,"Genomic_Coordinate"]))

#We are using hg19
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene

#promoter region set up to 1500, 500
promoter.re <- promoters(txdb,upstream=1500,downstream=500)


fi.cns <- c("TXID","TXCHROM","TXNAME","GENEID","TXSTART","TXEND","TXSTRAND")
txTable <- gsub(" ","",as.matrix(biomaRt::select(txdb,keys=names(promoter.re),columns=fi.cns,keytype="TXNAME")))
txTable <- txTable[!is.na(txTable[,2]),]

map.id <- getBM(attributes=c("hgnc_symbol","entrezgene_id"),filters="entrezgene_id", values=txTable[,"GENEID"],mart=ensembl)
mer.txTable <- unique(as.matrix(merge(txTable,map.id,by.x="GENEID",by.y="entrezgene_id")))
rownames(mer.txTable) <- mer.txTable[,"TXNAME"]


inter.pro.methyl <- as.matrix(findOverlaps(methyl.ran,promoter.re))
inter.pro.methyl <- inter.pro.methyl[is.element(names(promoter.re)[inter.pro.methyl[,2]],rownames(mer.txTable)),]
pro.met.re <- cbind(mer.txTable[names(promoter.re)[inter.pro.methyl[,2]],"hgnc_symbol"],sub_methyl_3[inter.pro.methyl[,1],c(1,3,4)])
colnames(pro.met.re) <- c("gene","methyl","chr","pos")
pro.met.re <- unique(pro.met.re[which(pro.met.re[,"gene"] != ""),])
#Final pro.met.re contains gene name, methylation id, chromosome, and position. 

```


## Differential methylation analysis

### Linear models using `limma`

In order to assess the association of tumor status with methylation values, we fit the following linear model for each gene $j=1,\dots,J$:
$$ Y_{ij} = \alpha_j + \beta_j X_{i} + \sum_{k=1}^K \gamma_{jk} Z_i + \epsilon_{ij} $$ where $i$ indexes patients, $Y_{ij}$ are methylation values for gene $j$ and patient $i$, $X_i$ is an indicator for being “with tumor,” $Z_i$ is a vector of patient level covariates, and $\epsilon_{ij} \sim N(0,\sigma^2)$. The patient level covariates included in the analysis were age and sex. The $J$ models were fit using empirical Bayes to allow for information sharing across genes. The R package `limma` was used to perform this analysis. The parameters of interest that give the association of tumor status with methylation were $\beta_j, j=1,\dots,J$. 

Since the goal was to identify genes with differential methylation by tumor status, hypothesis tests were performed to select genes with significant $\beta_j$ coefficients for the null hypothesis $\beta_j = 0$ against the alternative $\beta_j \neq 0$ for each gene $j$. The significance level was selected to be $\alpha=0.2$ and the Benjamini-Hochberg correction was used to correct for multiple comparisons and control the false discovery rate. 


### Bumphunter analysis - Differential methylation region

We used the averaged methylation value of all CpGs within the promoter
region of a gene to identify differential methylation between the tumor
existence in bladder cancer. Using the `bumphunter` function adapted from
the `minfi` R package, we obtained differential methylation regions
(DMRs). The Bumphunter function searches for DMRs by looking for GpGs
with mean differences [16]. The statistical model is:
$$Y_{ij}=\beta_{0}(l_{j}) +\beta_{1}(l_{j})X_{j}+\varepsilon_{ij}$$

where $i$ denotes the subject, $l_{j}$ represents the $j$th location, $X_{j}$ is the covariate of interest, $\varepsilon_{ij}$ denotes error, $\beta_{0}$ denotes baseline function and $\beta_{1}(l)$ is the parameter of interest. There are three steps: 1) creating clusters by t-statistical at each genomic location and a phenotype of interest, 2) breaking into segments by candidate region to be a cluster of probes, 3) test for the significance of the candidate region.  In this study, we used the covariates age and sex and the p-value adjusted FDR was below 0.05, and a change in methylation greater than 10% ($\Delta\beta > 0.1$) was used to identify a significant bump differential methylation region.



```{r message=FALSE}
#Bumphunter()

promoter_met_mat <- sub_methyl_3[(is.element(sub_methyl_3[,"Hybridization-REF"],pro.met.re[,"methyl"])),]
rownames(promoter_met_mat) <- promoter_met_mat[,1]
mat <- promoter_met_mat[,-c(1:4)] %>% as.matrix()
grset <- makeGenomicRatioSetFromMatrix(mat)
patient_coldata <- patient[is.element(patient[,21],tumor_status),]
patient_coldata <- patient_coldata[(!is.element(patient_coldata[,2],subset_nm)),]
colnames(patient_coldata) <- gsub("\\s", "", colnames(patient_coldata))
patient_coldata[,21] <- gsub("\\s", "", patient_coldata[,21])
patient_coldata$NoninvasiveBladderCancerTreatmentType %<>% gsub("\\(|)", "", .) %>% gsub("\\[|]","",.) %>% gsub("\\s", "",.)
colData(grset) <- cbind(colData(grset),patient_coldata[ order(match(patient_coldata[,2],rownames(colData(grset)))), ])
save(grset,file="grset")

status <- pData(grset)$PersonNeoplasmStatus
treatment <- pData(grset)$NoninvasiveBladderCancerTreatmentType
sex <-  pData(grset)$Sex
age <- pData(grset)$DiagnosisAge

deseignmatrix <- model.matrix(~status+sex+age)
#we use sex and age for covariates and 10% difference between the two groups

dmrs <- bumphunter(grset, design =deseignmatrix, type="Beta",cutoff=0.1)
save(dmrs,file="dmrs")
head(dmrs$table)
```


## Functional enrichment and network analysis

Functional enrichment analysis using ConsensusPathDb (Release35, CPDB:http://cpdb.molgen.mpg.de/CPDB) [17] was used to interpret the functional role of a gene identified with differential methylation status. Both significant pathway and gene ontology (GO) terms were defined as having an adjusted p-value ($\leq 0.05$). We constructed a protein-protein interaction (PPI) network for the significant methylated genes by string DB (Version 11.5) [18]. We selected the interaction relationship between genes with the highest confidence $> 0.9$. Pathway and GO term were visualized using the `ggplot2` R package (version 3.4.0).


# Results

## Differential DNA methylation analysis by tumor status

### Linear models using `limma`

```{r}
# more data cleaning

# subset methylation data according to pro.met.re
# and remove old, unnecessary columns
working.data = left_join(pro.met.re, sub_methyl_3, by=c('methyl'='Hybridization-REF'))
working.data = working.data %>% dplyr::select(-Gene_Symbol, -Chromosome, -Genomic_Coordinate)

# stash row information
meta.data = working.data[,1:4]

# average methylation by gene
working.data = working.data %>%
  dplyr::select(-methyl, -chr, -pos) %>%
  group_by(gene) %>%
  summarize_all(mean)

# load patient data
clinical.patient = read_delim('01.data/data_clinical_patient.txt', skip=4)
clinical.sample = read_delim('01.data/data_clinical_sample.txt', skip=4)

# combine the data and subset to those in working.data
patient = inner_join(clinical.patient, clinical.sample, by='PATIENT_ID')
patient = patient %>%
  filter(PATIENT_ID %in% colnames(working.data)) %>%
  group_by(PATIENT_ID) %>%  # one patient has two samples, remove second one
  dplyr::slice(n=1) %>%
  ungroup()
rm(list=c('clinical.patient', 'clinical.sample'))

# set column names lower case
colnames(patient) = tolower(colnames(patient))

# put patient data in same order as methylation data
patient = patient[match(colnames(working.data[,-1]), patient$patient_id), ]

```

```{r}

# Limma -------------------------------------------------------------------

# patient covariate matrix
design.matrix = model.matrix(~ tumor_status + sex + age,
                             data = patient)
colnames(design.matrix) = c('(Intercept)', 'tumor', 'male', 'age')

# fit models
limma.fit = lmFit(working.data[,-1], design=design.matrix)
limma.fit.eb = eBayes(limma.fit)

# look at results
top.table = topTable(limma.fit.eb, coef='tumor', sort.by='p', p.value=0.2, number=10000)

# extract gene names and meta information
genes.select = working.data$gene[as.numeric(row.names(top.table))]
genes.select.up = working.data$gene[as.numeric(row.names(top.table %>% filter(logFC>0)))]
genes.select.down = working.data$gene[as.numeric(row.names(top.table %>% filter(logFC<0)))]
genes.final = meta.data %>% filter(gene %in% genes.select)

top.table$gene = genes.select

# # save it
# write_csv(genes.final, 'diff_meth_genes.csv')

```

```{r, fig.cap='Figure 2a: Volcano plot for linear model analysis'}
# volcano plot

# get all results
all.res = topTable(limma.fit.eb, coef='tumor', sort.by='p', number=100000)
all.res$highlight = all.res$adj.P.Val < 0.2

# plot
gg_volcano = ggplot(all.res %>% mutate(P.Value = -log10(P.Value)), 
                    aes(x=logFC, y=P.Value, color=highlight)) +
  geom_point(size = 0.7) +
  scale_color_manual(values = c('black', 'blue'),
                     labels = c('No', 'Yes'),
                     name = 'Significant') +
  scale_x_continuous(limits = c(-0.1, 0.1)) +
  scale_y_continuous(limits = c(0, 6.2)) +
  labs(x = 'estimated beta coefficient',
       y = '-log10(p-value)') +
  theme(legend.position = 'right')

gg_volcano
```

Out of the 20,118 genes, 128 were identified to have differential methylation by tumor status. Most genes (119) had a negative coefficient indicating hypomethylation. Figure 2a above shows a volcano plot of the unadjusted p-values against estimated $\beta_j$ values, colored by significance. 

<!-- ```{r} -->
<!-- # pca -->
<!-- working.data.select = working.data %>% filter(gene %in% genes.select) -->
<!-- pc.0 = prcomp(t(working.data.select[,2:ncol(working.data.select)])) -->
<!-- dat = data.frame(pc.0$x[,1:2], tumor = patient$tumor_status) -->
<!-- ggplot(dat, aes(x=PC1, y=PC2, col=tumor)) + -->
<!--   geom_point() + -->
<!--   labs(title='pca on 128 identified genes') -->
<!-- ``` -->


The 128 identified genes were further subsetted to the genes that were also in the gene expression data. For the resulting 113 genes, Figure 2b below shows the average gene expression (log transformed) against estimated differential methylation values. There was a positive correlation (0.41), suggesting that at the gene level, larger differences in methylation (adjusted) were correlated with higher gene expression. These results may be un-intuitive since methylation is commonly associated with lower gene expression but these results may be due to lack of transcript-level data. It is also noteworthy that we are comparing gene expression with differences in methylation between tumor status groups, not methylation on the absolute scale.

```{r}
# read in gene expression data
mrna = read_delim('01.data/data_mrna_seq_v2_rsem.txt')

# subset to match working.data
mrna = mrna %>%
  dplyr::rename(gene = Hugo_Symbol) %>%
  dplyr::rename_all(function(x) gsub('-01$', '', x)) %>%
  dplyr::select_at(colnames(working.data))

# do log(1+x) transform
mrna[,2:ncol(mrna)] = log(1+mrna[,2:ncol(mrna)])

# # do pca
# pc.1 = prcomp(t(mrna[,2:ncol(mrna)]))
# dat = data.frame(pc.1$x[,1:2], tumor = patient$tumor_status)
# ggplot(dat, aes(x=PC1, y=PC2, col=tumor)) +
#   geom_point() +
#   labs(title='pca on transformed gene expression, all genes')
```


```{r}
# subset to match 128 identified genes
# only 113 in the dataset
mrna = mrna %>%
  filter(gene %in% genes.select)

# # pca
# pc.2 = prcomp(t(mrna[,2:ncol(mrna)]))
# dat = data.frame(pc.2$x[,1:2], tumor = patient$tumor_status)
# ggplot(dat, aes(x=PC1, y=PC2, col=tumor)) +
#   geom_point() +
#   labs(title='pca on transformed gene expression, 135 identified genes')
```

```{r, fig.cap='Figure 2b: Gene expression and estimated differential methylation according to linear model'}
# correlation b/t gene expression and diff methylation
mrna.avg = cbind(mrna$gene, rowMeans(mrna[,2:ncol(mrna)])) %>% as.data.frame()
top.table.2 = top.table %>%
  filter(gene %in% mrna$gene)
top.table.2 = top.table.2[match(mrna.avg$V1, top.table.2$gene), ]
# cor(top.table.2$logFC, as.numeric(mrna.avg$V2))

dat = data.frame(diff_meth = top.table.2$logFC, 
                 gene_exp = as.numeric(mrna.avg$V2))
ggplot(data=dat, aes(x=diff_meth, y=gene_exp)) +
  geom_point() +
  labs(x = 'Differential methylation (beta)',
       y = 'Average gene expression, log(1+x)')

```


### Bumphunter

A subset of 164,113 CpG methylation sites out of a total of 485,577 CpG sites was mapped to specific promoters (21,010 genes) for analysis. Differential methylation regions were examined in the individuals with a history of recurrence of tumor existence. The age and sex defined our study phenotype of interest. The 15 significant differential methylated regions were identified. With this region, there were predominately hypomethylated (n=13) in "With-Tumor" when compared to "Tumor-free."  This result corresponds to our `limma` significant differential methylation genes, and MOORE et al. [19] study, which showed global methylation in DNA from blood cells and leukocyte DNA hypomethylation was strongly associated with increased bladder cancer risk. In the DMR region, 15 genes were identified, and 14 were protein-coding genes. Eleven genes were strongly related to bladder cancer.  Nine genes overlap with the `limma` result. Figure 3 below shows a table of output from `bumphunter`.

<center>

![](image/table_bumphunter.png)

**Figure 3**: Bumphunter output
</center>

## Functional enrichment analysis and network

In an overrepresentation analysis of the combined set of both tools (`limma` and `bumphunter`), we identified 134 significant genes using ConsensusPathDB and revealed eight critical pathways (q-value < 0.1) (Figure 4a). This includes multiple vascular formation-related pathways:  Myogenesis, CDC42 signaling events, Apoptotic cleavage of cell adhesion protein, and VEGFA-VEGFR2. Within the set of 134 differentially methylated genes, 10 Gene Ontology (GO) pathways were significantly over-represented (q-value < 0.05), including "G-protein-coupled serotonin receptor activity" (GO:0004993), "serotonin receptor activity" (GO:0099589)," Thiamine metabolic process" (GO:0006772), and "protein-containing complex scaffold activity" (GO:0032947) (Figure 4b). The protein-protein interaction PPI) network constructed with the STRING database identified five genes with direct interaction (Figure 5). Four out of five genes are functionally grouped into the "Signaling by VEGF" pathway" (R-HSA-194138). The vascular endothelial growth factor (VEGF) pathway is highly associated with cancer that involves vasculogenesis and angiogenesis. Especially the most recent study reported that VEGF expression is associated with tumor grade and muscle-invasiveness of urinary bladder carcinoma when they evaluated immunohistochemical [20].

<!-- <center> -->

<!-- ![](image/enrichement.png) -->

<!-- **Figure 3**: Functional study of combined 134 genes -->

<!-- </center> -->


```{r message=FALSE}
#Finding genes in the DMR region
load("dmrs")
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
# load("ensembl")
bumphunter_gr <- GRanges(Rle(dmrs$table[,"chr"]),IRanges(dmrs$table[,"start"],dmrs$table[,"end"]+20000))
gene_list <- subsetByOverlaps(genes(txdb),bumphunter_gr)
map_ids <- getBM(attributes=c("hgnc_symbol","entrezgene_id"),filters="entrezgene_id", values=gene_list$gene_id,mart=ensembl)

map_ids #There are 15 genes identified in DMR region

#those are the genes that limma and bumphunter overlaps
intersect(top.table$gene,map_ids$hgnc_symbol)
```

```{r message=FALSE}
go <- fread("go.txt",sep ="\t", header = TRUE, stringsAsFactors = FALSE)
newgo <- cbind(go,do.call(rbind,lapply(strsplit(go$members_input_overlap_geneids,";") ,function(y)length(y))))
newgo$Generatio <- newgo$V1/newgo$size
newgo <- newgo  %>% as.data.frame

path <- fread("path.txt", sep ="\t", header = TRUE, stringsAsFactors = FALSE)
newpath <- cbind(path,do.call(rbind,lapply(strsplit(path$members_input_overlap_geneids,";") ,function(y)length(y))))
newpath$Generatio <- newpath$V1/newpath$size
 newpath$pathwaynm <- c(rep("patwhay"))
newpath <- newpath  %>% as.data.frame
```

```{r}
# create fake data
set.seed(1024) # keep reproducibility
go <- paste0("GO", sample(1000:2000, 5))
data <- data.frame("GOs" = rep(go, 2),
                   "Condition" = rep(c("A", "B"), each = 5),
                   "GeneRatio" = 1 / sample(10, 10),
                   "p.adjust" = 0.05 / sample(10, 10))
```

```{r, fig.cap='Figure 4a: Functional study of combined 134 genes'}
# plot: pathway

ggplot(newpath, aes(x = pathwaynm , y = pathway,
	color = as.double(newpath[,2]), size = as.double(newpath[,"Generatio"]))) +
	geom_point() +
	scale_color_gradient(low = "red", high = "blue") +
	theme_bw() +
	ylab("") +
  	xlab("") +
   	ggtitle("Pathway enrichment analysis")+
	labs(size="Generatio", colour="q-value")

```

```{r, fig.cap='Figure 4b: Functional study of combined 134 genes'}
# plot: Go
ggplot(newgo, aes(x = newgo[,"term_category"], y = newgo[,"term_name"],
	color = as.double(newgo[,2]), size = as.double(newgo[,"Generatio"]))) +
	geom_point() +
	scale_color_gradient(low = "red", high = "blue") +
	theme_bw() +
	ylab("") +
  	xlab("") +
   	ggtitle("GO enrichment analysis")+
	labs(size="Generatio", colour="q-value")
```

<center>

![](image/ppi.png)

**Figure 5**:First Protein-protein interaction network of combined 134 genes
</center>


## Case study

As described above, our differential methylated genes were enriched in pathways related to VEGF signaling. As a case study, we selected a gene for functional analysis, CDH5. CDH5 gene is known to play a role as cadherin and promotes tumor cell proliferation and invasion [21]. CDH5 is not yet studied deeply in bladder cancer, but this gene is a well-known biomarker in breast cancer cells, stimulates TGF-B signaling, and is associated with urinary exosomes [22]. In this study, we identified CDH5 as hypomethylated compared to the tumor-free group, and gene expression statistically increased.

<!-- <center> -->

<!-- ![](image/BCB784-Fall2022.png) -->

<!-- **Figure 4**:CDH5 -->
<!-- </center> -->


```{r, fig.cap='Figure 6a: CDH5'}
#geneset <- c("PRKCZ","CDH5","CTNNB1","APC2","TLE2") #This is the geneset that you want to print 
geneset <- c("CDH5") # Here, we want one. 
load("grset")
tumor_free <- assay(grset)[,which(grset$PersonNeoplasmStatus == "TUMORFREE" )]
withtumor <- assay(grset)[,which(grset$PersonNeoplasmStatus == "WITHTUMOR" )]
ppi_gene_mat <- promoter_met_mat[is.element(promoter_met_mat[,2],geneset),]
save(ppi_gene_mat,file="ppi_gene_mat")
load("ppi_gene_mat")

#Boxplot
for (i in 1:length(geneset)){
  onegene <- ppi_gene_mat[which(ppi_gene_mat[,2] == geneset[i]),]
  cpgid <- rownames(onegene)
  tfree_subset <- tumor_free[is.element(rownames(tumor_free),cpgid),]
  wtiht_subset <- withtumor[is.element(rownames(withtumor),cpgid),]
  rowmean_control <- cbind(t(rbind(rowMeans(tfree_subset))),rep("tumor_free"))
  rowmean_case <- cbind(t(rbind(rowMeans(wtiht_subset))),rep("with_tumor"))
  combine.mat <- rbind(rowmean_control,rowmean_case) %>% as.data.frame()
  colnames(combine.mat) <- c("mean_of_beta","group")
  p <- ggplot(combine.mat, aes(x=combine.mat$group, y=as.double(combine.mat$mean_of_beta),fill=group)) + geom_boxplot()+scale_fill_brewer(palette="Blues")+theme_classic() + labs(y= "Methlation_level (beta)", x = "Group") + labs(y= "Methlation_level (beta)", x = "Group",colour="Group")
  png(paste("03.plots","/",geneset[i],".png",sep=""))
  print(p)
  dev.off()
}

#This shows the plot for CDH5
ggplot(combine.mat, aes(x=combine.mat$group, y=as.double(combine.mat$mean_of_beta),fill=group)) + 
  geom_boxplot() +
  scale_fill_brewer(palette="Blues") +
  theme_classic() +
  labs(y= "Methlation_level (beta)", x = "Group",colour="Group")

```

```{r, fig.cap='Figure 6b: CDH5'}

#Correlation plot for CDH5
for (i in 1:length(geneset)){
  rsem_subset <- rsem[is.element(rsem[,1],geneset[i]),]
  rsem_test <- t(rsem_subset[-c(1:2)])
  onegene <- ppi_gene_mat[which(ppi_gene_mat[,2] == geneset[i]),]
  cpgid <- rownames(onegene)
  tfree_subset <- tumor_free[is.element(rownames(tumor_free),cpgid),]
  wtiht_subset <- withtumor[is.element(rownames(withtumor),cpgid),]
  colmean_control <- cbind(t(rbind(colMeans(tfree_subset))),rep("tumor_free"))
  colmean_case <- cbind(t(rbind(colMeans(wtiht_subset))),rep("with_tumor"))
  combine.mat <- rbind(colmean_control,colmean_case) %>% as.data.frame()
  combinedall <- merge(combine.mat,rsem_test,by='row.names',all.x=FALSE) %>% as.data.frame()
  colnames(combinedall) <- c("sample","beta","group","expression")
  p <- ggplot(combinedall, aes(x=as.double(combinedall$beta), y=as.double(combinedall$expression),color=combinedall$group)) + geom_point()+geom_smooth(method=lm, se=FALSE)+scale_color_brewer(palette="Paired")+theme_minimal() + labs(y= "Gene expression (TPM)", x = "Methlation_level (beta)",colour="Group")
  png(paste("03.plots","/",geneset[i],"corr",".png",sep=""))
  print(p)
  dev.off()
}

#This shows the plot for CDH5
ggplot(combinedall, aes(x=as.double(combinedall$beta), y=as.double(combinedall$expression),color=combinedall$group)) +
  geom_point() +
  geom_smooth(method=lm, se=FALSE) +
  scale_color_brewer(palette="Paired") +
  theme_minimal() +
  labs(y= "Gene expression (TPM)", x = "Methlation_level (beta)",colour="Group")

```


# Discussion

This study used two linear model algorithms (`limma` and `bumphunter`) to investigate differentially methylated genes in bladder recurrence. The analysis using gene-level linear models identified 128 genes, whereas the `bumphunter` R package identified 15 regions where most showed hypomethylation in both cases. Both hypomethylation and hypermethylation may occur in tumorigenesis bladder. In our case, hypomethylation was detected the most (90%). We expected oncogenes had been characterized; however, it was not correlated with gene expression statistically. There are several limitations to this result. First, the definitions of tumor status were coarse and may not adequately capture the biological constructs of interest. As noted earlier, tumor-free status describes a point in time; patients that are currently tumor-free may have a recurrence later. A time-to-event analysis that follows patients for a period of time and measures the durations of tumor and tumor-free status may be beneficial. Additionally, due to a lack of data availability, we did not consider cancer stage, tumor invasion, or treatment, which may affect the association of methylation and tumor status. 

Second, we used the RSEM gene expression data already pre-processed in cBio-Portal. Due to the nature of biology, defining promoters in this way may be inadequate since there are many isoforms, and the promoter can change locations. Therefore, we suggest further statistical analysis starting from raw data (ex., fastq/ bam) to extract counts in transcript level or transcription TPM value. This study was inherently exploratory since associations were examined. It is unclear how these results would generalize to other populations or if the association of methylation and tumor status is confounded by some variables not included in the analysis. 

We performed the gene enrichment analysis, and most genes were related to VEGF signaling pathways. VEGF pathway represents one of the promotes tumor angiogenesis cytokines and regulates the signal transmission for multiple cellular functions. VEGFs are also the most common class of therapeutic targets. Thus, pertinent to the findings of this study, VEGF plays multiple roles in cancer development and has been the focus of past studies to define differences in the reoccurrence of bladder cancer, primarily those originating only from NMIBC. Therefore, these alternative approaches and further studies may reveal more underlying molecular mechanisms contributing to bladder cancer recurrence. 


# Conclusion

High mutation rates in many cancer-related genes characterize bladder cancer, and the high recurrence rates are the most challenging for patient treatment. In this genome-wide profiling methylation study in recurrence bladder cancer, we found 134 differentially methylated genes in the recurrence status of bladder cancer. Over 90% of genes were hypomethylated, and these genes are highly associated with VEGF signaling pathways. Although most of our genes were negatively correlated with gene expression, further studies are necessary due to our limitations listed above. 


# References

[1]    C. Martin et al., “Familial Cancer Clustering in Urothelial Cancer: A Population-Based Case-Control Study,” J. Natl. Cancer Inst., vol. 110, no. 5, pp. 527–533, May 2018.

[2]    F. Turati et al., “Family history of cancer and the risk of bladder cancer: A case-control study from Italy,” Cancer Epidemiol., vol. 48, pp. 29–35, Jun. 2017.

[3]    E. Y.-W. Yu et al., “Family History and Risk of Bladder Cancer: An Analysis Accounting for First- and Second-degree Relatives,” Cancer Prev. Res. , vol. 15, no. 5, pp. 319–326, May 2022.

[4]    S. Li, M. R. Karagas, B. P. Jackson, M. N. Passarelli, and J. Gui, “Adaptive-mixture-categorization (AMC)-based g-computation and its application to trace element mixtures and bladder cancer risk,” Sci. Rep., vol. 12, no. 1, p. 17841, Oct. 2022.

[5]    D. S. Michaud, M. E. Wright, K. P. Cantor, P. R. Taylor, J. Virtamo, and D. Albanes, “Arsenic concentrations in prediagnostic toenails and the risk of bladder cancer in a cohort study of male smokers,” Am. J. Epidemiol., vol. 160, no. 9, pp. 853–859, Nov. 2004.

[6]    J. Dobruch et al., “Gender and Bladder Cancer: A Collaborative Review of Etiology, Biology, and Outcomes,” Eur. Urol., vol. 69, no. 2, pp. 300–310, Feb. 2016.

[7]    J. M. Sung et al., “Racial and Socioeconomic Disparities in Bladder Cancer Survival: Analysis of the California Cancer Registry,” Clin. Genitourin. Cancer, vol. 17, no. 5, pp. e995–e1002, Oct. 2019.

[8]    A. R. A. H. Hamid, F. R. Ridwan, D. Parikesit, F. Widia, C. A. Mochtar, and R. Umbas, “Meta-analysis of neoadjuvant chemotherapy compared to radical cystectomy alone in improving overall survival of muscle-invasive bladder cancer patients,” BMC Urol., vol. 20, no. 1, p. 158, Oct. 2020.

[9]    S. Byun et al., “Differential methylation of G-protein coupled receptor signaling genes in gastrointestinal neuroendocrine tumors,” Sci. Rep., vol. 11, no. 1, p. 12303, Jun. 2021.

[10]    M. Shivakumar, S. Han, Y. Lee, and D. Kim, “Epigenetic interplay between methylation and miRNA in bladder cancer: focus on isoform expression,” BMC Genomics, vol. 22, no. Suppl 3, p. 754, Oct. 2021.

[11]    M. Shivakumar, Y. Lee, L. Bang, T. Garg, K.-A. Sohn, and D. Kim, “Identification of epigenetic interactions between miRNA and DNA methylation associated with gene expression as potential prognostic markers in bladder cancer,” BMC Med. Genomics, vol. 10, no. Suppl 1, p. 30, May 2017.

[12]    X. Chen et al., “Urine DNA methylation assay enables early detection and recurrence monitoring for bladder cancer,” J. Clin. Invest., vol. 130, no. 12, pp. 6278–6289, Dec. 2020.

[13]    R. L. Grossman et al., “Toward a Shared Vision for Cancer Genomic Data,” New England Journal of Medicine, vol. 375, no. 12. pp. 1109–1112, 2016. doi: 10.1056/nejmp1607591.

[14]    J. Gao et al., “Integrative analysis of complex cancer genomics and clinical profiles using the cBioPortal,” Sci. Signal., vol. 6, no. 269, p. l1, Apr. 2013.

[15]    X. Shen, R. Liu, and M. Xing, “A six-genotype genetic prognostic model for papillary thyroid cancer,” Endocr. Relat. Cancer, vol. 24, no. 1, pp. 41–52, Jan. 2017.

[16]    A. E. Jaffe et al., “Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies,” Int. J. Epidemiol., vol. 41, no. 1, pp. 200–209, Feb. 2012.

[17]    A. Kamburov, U. Stelzl, H. Lehrach, and R. Herwig, “The ConsensusPathDB interaction database: 2013 update,” Nucleic Acids Res., vol. 41, no. Database issue, pp. D793–800, Jan. 2013.

[18]    D. Szklarczyk et al., “The STRING database in 2017: quality-controlled protein–protein association networks, made broadly accessible,” Nucleic Acids Res., vol. 45, no. D1, pp. D362–D368, Oct. 2016.

[19]    L. E. Moore et al., “Genomic DNA hypomethylation as a biomarker for bladder cancer susceptibility in the Spanish Bladder Cancer Study: a case-control study,” Lancet Oncol., vol. 9, no. 4, pp. 359–366, Apr. 2008.

[20]    A. Sadaf, M. Z. Rahman, P. Bhattacharjee, M. S. U. Ahamad, and S. Nasreen, “Significance of Vascular Endothelial Growth Factor Expression in the Bladder Urothelial Carcinoma and Its Association with Tumor Grade and Invasiveness,” Iran J Pathol, vol. 16, no. 4, pp. 362–369, Jul. 2021.

[21]    M. Inokuchi et al., “Cadherin 5 Is a Significant Risk Factor for Hematogenous Recurrence and a Prognostic Factor in Locally Advanced Gastric Cancer,” Anticancer Res., vol. 37, no. 12, pp. 6807–6813, Dec. 2017.

[22]    M. Telfah, R. A. Parikh, D. Zhang, and A. Kasi, “Metastatic Plasmacytoid Bladder Cancer Harboring a CDH-1 Mutation and Producing High Levels of CA 19-9. A Case Report and Literature Review,” Am. J. Case Rep., vol. 21, p. e923130, Jul. 2020.





